typescript/agents/customer_service/README.md

```diff
@@ -0,0 +1,94 @@
+
+**Note:** This agent is not integrated into an actual backend; behaviors are based on mocked tools defined in `customer_service/tools/tools.ts`.
+
+### Key Features
+
+- **Personalized Customer Assistance:** Greets returning customers and acknowledges purchase history.
+- **Product Identification:** Assists customers in identifying plants using visual aids (video/multimodal inputs).
+- **Order Management:** Accesses cart contents and modifies orders (add/remove items).
+- **Upselling:** Suggests relevant services (e.g., planting services) and handles pricing inquiries.
+- **Appointment Scheduling:** Checks availability and books slots for services.
+- **Customer Engagement:** Sends care instructions via SMS/Email and generates discount QR codes.
+
+#### Tools
+
+The agent has access to the following tools (defined in `customer_service/tools/tools.ts`):
+
+- `sendCallCompanionLink`: Sends a link for video connection to the user's phone.
+- `approveDiscount`: Approves a discount (within pre-defined limits).
+- `syncAskForApproval`: Requests discount approval from a manager.
+- `updateSalesforceCrm`: Updates customer records in Salesforce.
+- `accessCartInformation`: Retrieves the customer's cart contents.
+- `modifyCart`: Updates the customer's cart (add/remove items).
+- `getProductRecommendations`: Suggests suitable products based on plant type.
+- `checkProductAvailability`: Checks product stock at specific stores.
+- `schedulePlantingService`: Books a planting service appointment.
+- `getAvailablePlantingTimes`: Retrieves available time slots.
+- `sendCareInstructions`: Sends plant care information via email or SMS.
+- `generateQrCode`: Creates a discount QR code.
+
+## Setup and Installation
+
+### Prerequisites
+
+- Node.js (v20 or higher)
+- Google Cloud Project (for Vertex AI Gemini integration)
+
+### Installation
+
+1.  **Clone the repository** (if applicable) and navigate to the project directory:
+    ```bash
+    cd path/to/customer_service
+    ```
+
+2.  **Install dependencies**:
+    ```bash
+    npm install
+    ```
+    *Ensure `@google/adk`, `@google/genai`, `zod`, and `uuid` are in your `package.json`.*
+
+3.  **Set up Environment Variables**:
+
+    The agent requires Google Cloud credentials. You can set these in your shell or a `.env` file.
+    
+    Refer to `customer_service/config.ts` for the configuration loading logic.
+
+    ```bash
+    export GOOGLE_CLOUD_PROJECT=YOUR_PROJECT_ID
+    export GOOGLE_CLOUD_LOCATION=us-central1
+    export GOOGLE_GENAI_USE_VERTEXAI=1
+    export GOOGLE_API_KEY=MY_GOOGLE_API_KEY
+    ```
+
+## Running the Agent
+
+The entry point for the agent is `customer_service/agent.ts`. This file contains a `main()` function that initializes the agent, creates a session, and runs a sample interaction.
+
+To run the agent using `ts-node` (or `tsx`):
+
+```bash
+npx ts-node customer_service/agent.ts
```

Comment:
Currently when I run the agent with this command, it returns a response from the agent then exits.

Should we add `@google/adk-devtools` to `package.json` and change this command to something like the following to support interacting w/ the agent?

```
npx @google/adk-devtools run customer_service/agent.ts
```